Homework5/Task1SparseMatrix/SparseMatrix.cs

```diff
@@ -0,0 +1,96 @@
+﻿using System;
+using System.Collections;
+using System.Collections.Generic;
+using System.Linq;
+using System.Text;
+using System.Threading.Tasks;
+
+namespace Task1SparseMatrix
+{
+    class SparseMatrix: IEnumerable<int>
+    {
+        public int Rows { get; private set; }
+        public int Columns { get; private set; }
+        public Dictionary<int, int> MatrixElements { get; private set; } = new();
+
+        public SparseMatrix(int rows, int columns)
+        {
+            Rows = rows > 0 ? rows : 1;
+            Columns = columns > 0 ? columns : 1;
+        }
+
+        public int this[int row, int column]
+        {
+            get
+            {
+                int index = row * Columns + column;
+                int result = MatrixElements.TryGetValue(index, out result) ? result : default;
+                return result;
+            }
+
+            set
+            {
+                int index = row * Columns + column;
+                MatrixElements[index] = value;
```

Comment:
Consider sparse matrix sized 1mln x 1mln with single non-null element at [0,0] index.
It takes 4tb space and keeps single number.

Do not keep zeroes